core/application/impl/config_reader/pt_util.hpp

```diff
@@ -6,25 +6,20 @@
 #ifndef KAGOME_APPLICATION_UTIL_HPP
 #define KAGOME_APPLICATION_UTIL_HPP
 
-#include <boost/property_tree/ptree.hpp>
-#include <outcome/outcome.hpp>
-
+#include "application/impl/config_reader/error.hpp"
 #include "application/impl/kagome_config.hpp"
 
 namespace kagome::application {
 
-  /**
-   * Initialise kagome config parameters from a boost property tree
-   */
-  outcome::result<KagomeConfig> initConfigFromPropertyTree(
-      const boost::property_tree::ptree &tree);
+  template <typename T>
+  outcome::result<std::decay_t<T>> res(boost::optional<T> opt_entry) {
```

Comment:
I don't like this name. `res` has too common meaning. Maybe `check` or `ensure`?